File: Codeforces/CampingWinter2020ESCOM/Day3/RemaindersGame.cpp
#include <iostream>
#include <numeric>

using namespace std;
using num = uint_fast64_t;

auto lcm(num a, num b) { return a * b / gcd(a, b); }
auto main() -> int {
  ios::sync_with_stdio(false);
  cin.tie(nullptr);

  auto n = num {}, k = num {};
  cin >> n >> k;

  auto lcm_total = num {1}, buffer = num {};
  while (n-- and lcm_total) {
    cin >> buffer;
    lcm_total = lcm(lcm_total, buffer) % k;
  }

  cout << (lcm_total ? "No" : "Yes") << endl;

  return 0;
}

File: Codeforces/CampingWinter2020ESCOM/Day3/RemaindersGame.py
import math
n, k = map(int, input().split())
numbers = list(map(int, input().split()))

lcm = lambda a, b: a * b // math.gcd(a, b)

lcm_total = 1
for n in numbers: lcm_total = lcm(lcm_total, n) % k

print ("Yes" if lcm_total % k == 0 else "No")